### PR TITLE
OCPBUGS-34829: fix issuer check during JWT authentication 4.13

### PR DESCRIPTION
**Pull request by thibaultmg**

### CODE REVIEW

**Comment by juzhao** [comment]
it seems can not get the response for the prow job, same `Waiting for status to be reported` message yesterday

File: pkg/server/requestlog.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			start := time.Now()
+
+			// Wrap the responseWriter to capture the HTTP status code.
+			aw := &AccessLogResponseWriter{w, http.StatusOK}
+
+			next.ServeHTTP(aw, r)
+
+			level.Info(logger).Log(
```
Comment (juzhao) [inline pkg/server/requestlog.go:34]: ci/prow/unit failed
https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_telemeter/540/pull-ci-openshift-telemeter-release-4.13-unit/1807616940810702848
```
 pkg/server/requestlog.go:34:26: Error return value of `(github.com/go-kit/log.Logger).Log` is not checked (errcheck)
			level.Info(logger).Log( 
```

**Comment by juzhao** [comment]
ci/prow/unit failed
https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_telemeter/540/pull-ci-openshift-telemeter-release-4.13-unit/1807616940810702848
```
 test/e2e/receive_test.go:150:6: too many arguments in call to receiver.LimitBodySize
	have ("github.com/go-kit/kit/log".Logger, number, http.HandlerFunc)
	want (int64, http.Handler) (typecheck) 
```
see code, there are 3 parameters
https://github.com/openshift/telemeter/blob/release-4.13/test/e2e/receive_test.go#L149-L151

**Comment by simonpasquier** [comment]
@thibaultmg the patch would need a rework.

**Pull request merged**